core/src/system_monitor_service.rs

```diff
@@ -690,6 +695,112 @@ impl SystemMonitorService {
         })
     }
 
+    /*
+    fn string_append_u32_as_char(&mut string: String, val: u32)
+    {
+        let mut v = val;
+
+        for _i in 0..4 {
+            string.push(v as u8 as char);
+            v >>= 8;
+        };
+    }*/
+
+    #[cfg(any(target_arch = "x86_64", target_arch = "x86_64"))]
+    unsafe fn report_cpuid_values() {
+        let cpuid_0 = __cpuid(0);
+
+        let max_leaf = cpuid_0.eax;
+        let max_subleaf_7 = if 7 <= max_leaf {
+            __get_cpuid_max(7).1
+        } else {
+            0
+        };
+
+        let mut manufacturer = String::from("");
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ebx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.edx >> (i * 8)) as u8 as char);
+        }
+        for i in 0..4 {
+            manufacturer.push((cpuid_0.ecx >> (i * 8)) as u8 as char);
+        }
+
+        let cpuid_1 = if 1 <= max_leaf {
+            __cpuid(1)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_2 = if 2 <= max_leaf {
+            __cpuid(2)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_4 = if 4 <= max_leaf {
+            __cpuid(4)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_7_0 = if 7 <= max_leaf {
+            __cpuid_count(7, 0)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+        let cpuid_7_1 = if 7 <= max_leaf && 1 <= max_subleaf_7 {
+            __cpuid_count(7, 1)
+        } else {
+            CpuidResult {
+                eax: 0,
+                ebx: 0,
+                ecx: 0,
+                edx: 0,
+            }
+        };
+
+        datapoint_info!(
+            "cpuid-values",
+            ("manufacturer_id", manufacturer, String),
```

Comment:
we can recover the manufacturer from one of the other registers, right?  strings in metrics are kinda pricey

Reply:
I don't think you can recover mfr from other registers. If we think there are only a few things we care about, could make a little hashtable to map string to an internal enum, if you think that's worthwhile.

Reply:
I'm combining things.. Really we're only ever going to see GenuineIntel or AuthenticAMD, so I'm just adding an explicit test for these without even converting to a string. It'll be Intel, AMD, or Other. We can expand "Other" if it's useful in the future.